authz/multitenant_client.go

```diff
@@ -21,20 +22,21 @@ import (
 )
 
 var (
-	ErrMissingConfig  = errors.New("missing config")
-	ErrMissingStackID = status.Errorf(codes.InvalidArgument, "missing stack ID")
-	ErrMissingAction  = status.Errorf(codes.InvalidArgument, "missing action")
-	ErrMissingCaller  = status.Errorf(codes.Unauthenticated, "missing caller")
-	ErrMissingSubject = status.Errorf(codes.Unauthenticated, "missing subject")
-	ErrReadPermission = status.Errorf(codes.PermissionDenied, "read permission failed")
+	ErrMissingConfig    = errors.New("missing config")
+	ErrMissingNamespace = status.Errorf(codes.InvalidArgument, "missing namespace")
+	ErrInvalidNamespace = status.Errorf(codes.InvalidArgument, "invalid namespace")
+	ErrMissingAction    = status.Errorf(codes.InvalidArgument, "missing action")
+	ErrMissingCaller    = status.Errorf(codes.Unauthenticated, "missing caller")
+	ErrMissingSubject   = status.Errorf(codes.Unauthenticated, "missing subject")
+	ErrReadPermission   = status.Errorf(codes.PermissionDenied, "read permission failed")
 )
 
 type CheckRequest struct {
-	Caller     claims.AuthInfo
-	StackID    int64
-	Action     string
-	Resource   *Resource
-	Contextual []Resource
+	Caller    claims.AuthInfo
+	Namespace string
+	Action    string
+	Object    string
```

Comment:
Should I split this:
```suggestion
	// ~Kind eg dashboards
	Resource string
	// The specific resource
	// In grafana, this was historically called "UID", but in k8s, it is the name
	Name string
```